packages/toolkit/src/extensionShared.ts

```diff
@@ -238,6 +234,36 @@ function logAndShowWebviewError(err: unknown, webviewId: string, command: string
     })
 }
 
+/**
+ * Returns an object that provides AWS service endpoints that the toolkit supports.
+ * 
+ * https://docs.aws.amazon.com/general/latest/gr/rande.html
+ */
+function makeEndpointsProvider() {
+    let localManifestFetcher: ResourceFetcher
+    let remoteManifestFetcher: ResourceFetcher
+    if (isWeb()) {
+        // In web mode everything must be in a single file, so things like the endpoints file will not be available.
+        // The following imports the endpoints file, which causes webpack to bundle it in the final output file
+        const endpoints = require('../resources/endpoints.json')
+        localManifestFetcher = { get: async () => JSON.stringify(endpoints) }
+        // Cannot use HttpResourceFetcher due to web mode breaking on import
```

Comment:
we plan to fix that, right?
```suggestion
        // TODO: HttpResourceFetcher is currently broken in web mode.
```